examples/with-nextjs-and-clerk-auth/src/lib/monite-api/demo-data-generator/bank-account.ts

```diff
@@ -17,38 +21,54 @@ export const createBankAccount = async ({
   is_default_for_currency: true;
   entity_id: string;
 }) => {
-  const { POST } = createMoniteClient({
-    headers: {
-      Authorization: `${token.token_type} ${token.access_token}`,
-    },
-  });
+  const { POST, getEntity } = createMoniteClient(token);
+
+  const entity = await getEntity(entity_id);
 
-  const display_name = faker.company.name();
-  const country = getRandomItemFromArray(['GE', 'DE', 'GB'] satisfies Array<
-    components['schemas']['AllowedCountries']
-  >);
+  const entityCountry = entity.address.country;
+  const country: keyof typeof bankCountriesToCurrencies = (
+    entityCountry in bankCountriesToCurrencies
+      ? entityCountry
+      : getRandomCountry()
+  ) as keyof typeof bankCountriesToCurrencies;
 
+  const bankName = faker.company.name();
+  const currency = bankCountriesToCurrencies[country];
+  const accountCreationParams = {
+    is_default_for_currency,
+    bank_name: `${bankName} Bank`,
+    display_name: faker.finance.accountName(),
+    currency: currency,
+    country,
+  } as components['schemas']['CreateEntityBankAccountRequest'];
```

Comment:
If the type changes, the TypeScript will throw "conversion may be a mistake because neither type sufficiently overlaps with the other". 

I intentionally added "as" here to perform a static type check. Older code provided "any" object as a payload to web request without performing any type checking according to schema we have.